scripts/setup-vagrant.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+case $ST2_TARGET in
+  "el7")
+    DC_TARGET=centos7
+    INSTALL_CMD="yum";;
+  "trusty"|"xenial")
+    DC_TARGET=$ST2_TARGET
+    INSTALL_CMD="apt-get";;
+  *)
+    echo "[Error] Unknown target $ST2_TARGET"
+    exit 1;;
+esac
+
+echo "[Install] dependencies"
+sudo $INSTALL_CMD update
+if [[ $ST2_TARGET != 'el7' ]]; then
+  sudo apt-get -y autoremove
+fi
+
+sudo $INSTALL_CMD install -y linux-image-extra-$(uname -r)
+sudo $INSTALL_CMD install -y git curl wget
+
+# Install docker-compose
+DC_BIN="/usr/local/bin/docker-compose"
+DC_URL="https://github.com/docker/compose/releases/download/1.9.0/docker-compose-`uname -s`-`uname -m`"
+if [[ ! -x $DC_BIN ]]; then
+  echo "[Install] docker-compose $ST2_TARGET"
+  sudo sh -c "curl -sL $DC_URL > $DC_BIN"
+  sudo chmod +x $DC_BIN
+fi
+
+# Using docker-compose, 1) build packages, and 2) run quick tests
+if [[ "${ST2_GITURL}" != "" ]]; then
+  ST2URL="-e ST2_GITURL=$ST2_GITURL"
+fi
+if [[ "${ST2_GITREV}" != "" ]]; then
+  ST2REV="-e ST2_GITREV=$ST2_GITREV"
+fi
+if [[ "${ST2MISTRAL_GITURL}" != "" ]]; then
+  ST2MISTRALURL="-e ST2MISTRAL_GITURL=$ST2MISTRAL_GITURL"
+fi
+if [[ "${ST2MISTRAL_GITREV}" != "" ]]; then
+  ST2MISTRALREV="-e ST2MISTRAL_GITREV=$ST2MISTRAL_GITREV"
+fi
+sudo sh -c "(cd /vagrant && $DC_BIN run $ST2URL $ST2REV $ST2MISTRALURL $ST2MISTRALREV --rm $ST2_TARGET)"
+
+if [ "$ST2_INSTALL" = "yes" ]; then
+  echo 'Install st2 packages'
+
+  # Halt the docker test environment (otherwise, the subsequent self-verification will fail)
+  sudo docker stop "vagrant_${DC_TARGET}test_1"
+
+  # Install the packages we just built
+  if [[ $ST2_TARGET != 'el7' ]]; then
+    sudo sh -c '(cd /tmp/st2-packages && dpkg -i st2*.deb)'
```

Comment:
I'll look into `gdepi`. Thanks for the suggestion. Everything seems to work as is.. but I would expect `gdepi` to allow me to clean things up.